pkg/builder/step_patch_ld_preload.go

```diff
@@ -0,0 +1,41 @@
+package builder
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type stepPatchLdPreload struct {
+	ChrootKey             string
+}
+
+func (s *stepPatchLdPreload) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	// Read our value and assert that it is they type we want
+	chrootDir := state.Get(s.ChrootKey).(string)
+
+	ui := state.Get("ui").(packer.Ui)
+	ui.Say("Patching ld.preload.so for missing libarmmem")
+	ldPreload := filepath.Join(chrootDir, "/etc/ld.so.preload")
+	out, err := exec.Command("sed", "-e", "/libarmmem/ s/^#*/#/", "-i", ldPreload).CombinedOutput()
+        if err != nil {
+                ui.Error(fmt.Sprintf("error sed -e '/libarmmem/ s/^#*/#/' -i %s %v: %s", ldPreload, err, string(out)))
+                s.Cleanup(state)
+                return multistep.ActionHalt
+        }
+	out, err = exec.Command("cat", ldPreload).CombinedOutput()
+        if err != nil {
+                ui.Error(fmt.Sprintf("error cat %s %v: %s", ldPreload, err, string(out)))
+                s.Cleanup(state)
+                return multistep.ActionHalt
+        }
+	ui.Message(fmt.Sprintf(string(out)))
+	return multistep.ActionContinue
+}
+
+func (s *stepPatchLdPreload) Cleanup(state multistep.StateBag) {
```

Comment:
Shouldn't you remove the modifications to `/etc/ld.so.preload` afterwards? Maybe it would be cleaner to bind mount `/dev/null` to `/etc/ld.so.preload`?

Reply:
I agree it would be cleaner and easier to bind mount `/etc/ld.so.preload` to `/dev/null` in this specific case.
Do we know what could be in the `/etc/ld.so.preload` file besides this buggy declaration?